verilog/CST/functions_test.cc

```diff
@@ -388,5 +391,59 @@ TEST(GetFunctionFormalPortsGroupTest, WithFormalPorts) {
   }
 }
 
+TEST(GetFunctionHeaderTest, GetFunctionName) {
+  constexpr int kTag = 1;  // value doesn't matter
+  const SyntaxTreeSearchTestCase kTestCases[] = {
+      {""},
+      {"module m(); endmodule: m"},
+      {"function ",
+       {kTag, "foo"},
+       "();\n endfunction\n function ",
+       {kTag, "bar"},
+       "()\n; endfunction"},
+      {"function int ",
+       {kTag, "foo"},
+       "();\n endfunction\n function ",
+       {kTag, "bar"},
+       "()\n; endfunction"},
+      {"function int ",
+       {kTag, "foo"},
+       "(int a, bit x);\nreturn a;\n endfunction\n function ",
+       {kTag, "bar"},
+       "()\n; endfunction"},
+
+      {"module my_module;\nfunction int ",
+       {kTag, "inner_function"},
+       "(int my_args);\nreturn my_args;\nendfunction\nendmodule"},
+      {"class function_class;\nfunction int ",
```

Comment:
Btw, you can always check https://github.com/google/verible/blob/master/verilog/parser/verilog_parser_unittest.cc for more ideas.

Thought of some more that are common:

* `virtual` functions inside classes, include `pure virtual`
* `static` functions inside classes (you've covered `automatic` already)
* `import "DPI-C"` functions (see LRM 35.5.4 Import declarations). Or file a TODO for these for later because they may require a little more work.